platform_design/lib/tab_1.dart

```diff
@@ -0,0 +1,161 @@
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+
+import 'tab_1_detail.dart';
+import 'utils.dart';
+import 'widgets.dart';
+
+const tab1Title = 'Songs';
+const tab1AndroidIcon = Icon(Icons.music_note);
+const tab1IosIcon = Icon(CupertinoIcons.music_note);
+const _itemsLength = 50;
+
+class Tab1 extends StatefulWidget {
+  const Tab1({ Key key, this.androidDrawerBuilder, this.platformOverride })
+      : super(key: key);
+
+  final WidgetBuilder androidDrawerBuilder;
+  final ValueNotifier<TargetPlatform> platformOverride;
+
+  @override
+  _Tab1State createState() => _Tab1State();
+}
+
+class _Tab1State extends State<Tab1> {
+  final _androidRefreshKey = GlobalKey<RefreshIndicatorState>();
+
+  List<MaterialColor> colors;
+  List<String> songNames;
+
+  @override
+  void initState() {
+    _setData();
+    super.initState();
+  }
+
+  void _setData() {
+    colors = getRandomColors(_itemsLength);
+    songNames = getRandomNames(_itemsLength);
+  }
+
+  Future<void> _refreshData() {
```

Comment:
Not sure I understood what you meant. Where is would the setState be?

Reply:
I'm still not sure I understand. The 2s wait still needs to be somewhere (so the user has something to look at as it performs a fake refresh). The getRandomColors from utils.dart shouldn't just return a Future that waits 2s since we don't know where that function is going to be called etc.